src/client/client.ts

```diff
@@ -272,6 +272,19 @@ export class Client {
       baseUrl: config.baseUrl ? config.baseUrl.replace(/\/$/, "") : "https://qstash.upstash.io",
       authorization: `Bearer ${config.token}`,
     });
+
+    if (!config.token) {
+      throw new Error(
+        "QStash token is required!\n\n" +
+          "To fix this:\n" +
+          "1. Get your token from the Upstash Console (https://console.upstash.com/qstash)\n" +
+          "2. Initialize the client with:\n\n" +
+          "   const client = new Client({\n" +
+          "     token: '<YOUR_QSTASH_TOKEN>'\n" +
+          "   });"
```

Comment:
we had issues with throwing in the constructor if the token is missing in redis. more info available here https://github.com/upstash/redis-js/pull/1311

I would prefer if we don't throw but simply log a warning

Reply:
what was the issue? Afaik stripe also throws and their dx is incredible, logging is a bit harder to see

Reply:
if redis-js is used in turbo repo, it gets an error at compile time unless the env upstash env variables are specified in the turbo.json file.